### PR TITLE
style(ComboBox): add animation

### PR DESCRIPTION
**Pull request by nikgraf**
@jpuri very subtle animations :)

### CODE REVIEW

File: src/style/rating.js [hunk 1]
```diff
@@ -50,7 +50,12 @@ const ratingStyle = {
...
   characterStyle: {
     color: '#FFCC00',
-    textShadow: '0px 1px 0px #DCB000'
+    textShadow: '0px 1px 0px #DCB000',
+    top: 0,
+
+    /* animations */
+    transition: 'color 0.1s',
+    transitionTimingFunction: 'ease-out'
   },
```
Comment (jpuri) [inline src/style/rating.js]: @nikgraf , should we not add this to activeCharacterStyle and probably hoverCharacterStyle also.

File: src/style/button.js [hunk 2]
```diff
@@ -17,6 +17,10 @@ const buttonStyle = {
     textDecoration: 'none',
```
Comment (jpuri) [inline src/style/button.js]: In button this transition is sort of giving me feeling of delay when I click.... coz click is such a fast action.

File: src/style/toggle.js [hunk 3]
```diff
@@ -148,7 +152,8 @@ const toggleStyle = {
     height: 28,
     backgroundColor: 'rgb(246, 246, 246)',
     boxShadow: '0 0 0 0 rgb(189, 189, 189)',
-    cursor: 'url(http://www.google.com/intl/en_ALL/mapfiles/closedhand.cur), move'
+
```
Comment (jpuri) [inline src/style/toggle.js]: +1, it was so touch to get that grabbing hand to perfection.

File: src/style/toggle.js [hunk 4]
```diff
@@ -48,7 +49,10 @@ const toggleStyle = {
     cursor: 'pointer',
     border: '1px solid rgb(220, 220, 220)',
     boxShadow: '0 1px 0px 0px rgb(185, 185, 185)',
-    transition: 'left 0.1s ease-in-out',
+
+    /* animations */
+    transition: 'left 0.1s, background-color 0.1s, boxShadow 0.1s, height 0.1s',
+    transitionTimingFunction: 'ease-in-out',
```
Comment (jpuri) [inline src/style/toggle.js]: This animation of left on handle looks good when you tap....but looks little broken when you grab (at least on my machine).

File: src/style/toggle.js [hunk 5]
```diff
@@ -26,7 +26,8 @@ const toggleStyle = {
     position: 'relative',
     // Calculated with 2 * the width of choice area
     width: 92,
-    transition: 'left 0.1s ease-in-out'
+    transition: 'left 0.1s',
+    transitionTimingFunction: 'ease-in-out'
   },
```
Comment (jpuri) [inline src/style/toggle.js]: Why is animation on left needed for slider ?

File: src/style/rating.js (hunk 1 above)
Comment (nikgraf) [inline src/style/rating.js]: focusStyle & hoverStyle inherit from style :smile: 

This way you only have to add it once. Less code & same functionality.

File: src/style/toggle.js (hunk 5 above)
Comment (nikgraf) [inline src/style/toggle.js]: The left animation is the one when just clicking on the toggle.

File: src/style/toggle.js (hunk 4 above)
Comment (jpuri) [inline src/style/toggle.js]: Transition on toggle at times is giving me feeling that grab is not smooth any more.

File: src/style/rating.js (hunk 1 above)
Comment (jpuri) [inline src/style/rating.js]: :)

File: src/style/toggle.js (hunk 5 above)
Comment (jpuri) [inline src/style/toggle.js]: we need it for handle....but slider does not even have 'left' changing.

**Comment by jpuri** [comment]
:+1:

**Pull request merged**